shap/explainers/tree.py

```diff
@@ -65,10 +66,12 @@ def __init__(self, model, **kwargs):
             self.trees = model.get_booster()
         elif str(type(model)).endswith("lightgbm.basic.Booster'>"):
             self.model_type = "lightgbm"
-            self.trees = model
+            self.model = model
+            self.trees = [Tree(e,num_trees=len(model.dump_model()["tree_info"])) for e in model.dump_model()["tree_info"]]
```

Comment:
Seems strange for a tree to need to know how many trees there are. Perhaps we could save this info elsewhere? Or just change it to a `rescale` parameter?

Reply:
This is necessary to rescale `self.values` by the number of trees for lightGBM. Since currently only the current tree is passed into the tree init function, there isn't a great way of having a new parameter other than just passing it to the function.